### DISCUSSION TITLE
Multiplexing configuration load order?

### DISCUSSION THREAD

**Discussion by dongdongli-g** [wezterm/wezterm#2657] [Q&A]
With multiplexing especially connect to remote ssh. For example:

If I have `wezterm connect server`, it pops up a new window. On my server, I don't have $HOME/.wezterm.lua or $HOME/.config/wezterm.

Some observation:
1. It respects my local wezterm config like shortcuts 
2. It seems like first tab theme is correct. But when I open a second tab, the theme doesn't seem correct. 

From the [configuration manual](https://wezfurlong.org/wezterm/config/files.html), it describes how local wezterm configuration file works. But I am confused how it works for multiplexing. Some questions here:

1. Do I only local wezterm config for multiplexing? For example, theming, shortcuts, scrollback buffer, copy mode and etc..
2. What is the configuration file load order if I have local:$HOME/.wezterm.lua and server:$HOME/.wezterm.lua?

**Comment by yutkat** [comment]
I think it's currently impossible to load server-side settings. I solve this situation by switching the settings on the local side.
I know three solutions.

### 1 Switching by local environment variables

https://github.com/yutkat/dotfiles/blob/c1f642b22d7f7b1b98a28a8a1497249a54ca97e9/.config/wezterm/on.lua#L69-L86
https://github.com/yutkat/dotfiles/blob/c1f642b22d7f7b1b98a28a8a1497249a54ca97e9/.config/zsh/rc/function.zsh#L674-L686
ref: https://github.com/wez/wezterm/issues/1680

### 2 Merge configuration files on the local side

https://github.com/yutkat/dotfiles/blob/c1f642b22d7f7b1b98a28a8a1497249a54ca97e9/.config/wezterm/wezterm.lua#L42-L56

### 3 Switching the config using `--config-file` as startup argument for multiplexer

```lua
-- /tmp/wezterm.lua
return {
	ssh_domains = {
		{
			name = "my.server",
			remote_address = "192.168.1.100",
			username = "aaa",
		},
	},
	color_scheme = "Batman",
}

```

`wezterm --config-file=/tmp/wezterm.lua connect my.server`

**Comment by wez** [comment]
The server and client machines have their own distinct configurations.
The GUI doesn't read the configuration from the server, and vice versa.

However, since the color palette for a pane is controlled through escape sequences, a pane that was spawned on the server will read the color scheme from the server to set up the starting palette at the time that it is spawned.  That palette is sent to the client when it connects in order for it to correctly reflect the palette of the pane, and is updated whenever escape sequences change the palette.  This isn't synchronization of a configuration option, it's reporting a property of the terminal emulator in the pane.

This issue is open to improve the documentation around this:
* https://github.com/wez/wezterm/issues/1268

In general, you want your configuration to reside in the config file of the GUI/client side: that's where most of the config options apply and will make the most difference.

What I do is sync the same config to both sides using eg: a git dotfiles repo, so both sides have the same config.
